extension/filetemplateextension/extension.go

```diff
@@ -0,0 +1,79 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package filetemplateextension // import "github.com/elastic/opentelemetry-collector-components/extension/filetemplateextension"
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+
+	"github.com/elastic/opentelemetry-collector-components/pkg/templates"
+)
+
+type fileTemplateExtension struct {
+	config *Config
+}
+
+var _ templates.TemplateFinder = &fileTemplateExtension{}
+
+func newFileTemplateExtension(config *Config) *fileTemplateExtension {
+	return &fileTemplateExtension{
+		config: config,
+	}
+}
+
+func (e *fileTemplateExtension) FindTemplate(ctx context.Context, name, version string) (templates.Template, error) {
+	path := filepath.Join(e.config.Path, name+".yml")
```

Comment:
Should we rely on the user's path instead? I would expect defining the extension of the file too when using the `file extension`. 

For example, I initially defined my template as a "httpcheck.yaml" file, and I was getting this error:

```
Error: cannot start pipelines: failed to find template "httpcheck": not found
2024/08/21 08:56:04 collector server run finished with error: cannot start pipelines: failed to find template "httpcheck": not found
```

```suggestion
	path := filepath.Join(e.config.Path, name)
```

Reply:
I think it is better to use a "name" in the configuration, so it is the same independently of the source of the templates, a source could be a web service, and there might be no extensions there.

Something we could do though is to accept multiple extensions here if it is going to make things easier for users.